zenoh/src/net/routing/interceptor/access_control.rs

```diff
@@ -0,0 +1,282 @@
+                        }
+                    }
+                }
+            }
+        }
+        let policy_enforcer = self.enforcer.clone();
+        (
+            Some(Box::new(IngressAclEnforcer {
+                policy_enforcer: policy_enforcer.clone(),
+                interface_list: interface_list.clone(),
+                default_decision: match policy_enforcer.default_permission {
+                    Permission::Allow => true,
+                    Permission::Deny => false,
+                },
+                zid,
+            })),
+            Some(Box::new(EgressAclEnforcer {
+                policy_enforcer: policy_enforcer.clone(),
+                interface_list,
+                default_decision: match policy_enforcer.default_permission {
+                    Permission::Allow => true,
+                    Permission::Deny => false,
+                },
+                zid,
+            })),
+        )
+    }
+
+    fn new_transport_multicast(
+        &self,
+        _transport: &TransportMulticast,
+    ) -> Option<EgressInterceptor> {
+        None
+    }
+
+    fn new_peer_multicast(&self, _transport: &TransportMulticast) -> Option<IngressInterceptor> {
+        None
+    }
+}
+
+impl InterceptorTrait for IngressAclEnforcer {
+    fn compute_keyexpr_cache(&self, key_expr: &KeyExpr<'_>) -> Option<Box<dyn Any + Send + Sync>> {
+        Some(Box::new(key_expr.to_string()))
+    }
+    fn intercept<'a>(
+        &self,
+        ctx: RoutingContext<NetworkMessage>,
+        cache: Option<&Box<dyn Any + Send + Sync>>,
+    ) -> Option<RoutingContext<NetworkMessage>> {
+        let key_expr = cache
+            .and_then(|i| i.downcast_ref::<String>().map(|e| e.as_str()))
+            .or_else(|| ctx.full_expr())?;
+        if let NetworkBody::Push(Push {
+            payload: PushBody::Put(_),
+            ..
+        })
+        | NetworkBody::Request(Request {
+            payload: RequestBody::Put(_),
+            ..
+        })
+        | NetworkBody::Response(Response {
+            payload: ResponseBody::Put(_),
+            ..
+        }) = &ctx.msg.body
+        {
+            let mut decision = self.default_decision;
+
+            for subject in &self.interface_list {
+                match self.policy_enforcer.policy_decision_point(
+                    *subject,
+                    Action::Put,
+                    key_expr,
+                    self.default_decision,
+                ) {
+                    Ok(true) => {
+                        decision = true;
+                        break;
+                    }
+                    Ok(false) => continue,
+                    Err(e) => {
+                        log::error!("Authorization incomplete due to error {}", e);
+                        return None;
+                    }
+                }
+            }
+
+            if !decision {
+                log::warn!("{} is unauthorized to Put", self.zid);
```

Comment:
This should be `debug` at most. Every warning may be logged and could cause the log system to be understrasse.